SpamPkg/Tools/GenSpamArtifacts/GenSpamArtifacts.py

```diff
@@ -0,0 +1,203 @@
+    def generate_spam_artifacts(aux_config_path: Path, mm_supervisor_build_dir: Path, stm_build_dir: Path, output_dir: Path):
+        """Generates SPAM artifacts.
+
+        Generates the following artifacts:
+        - MmSupervisorCore.aux
+        - MmSupervisorCore.efi
+        - Stm.bin
+
+        Args:
+            aux_config_path: Path to the aux gen config file.
+            mm_supvervisor_build_dir: Path to the MM Supervisor build output.
+            stm_build_dir: Path to the STM build output.
+            output_dir: Path to place the artifacts.
+        """
+        try:
+            aux_path = generate_aux_file(aux_config_path, mm_supervisor_build_dir, output_dir)
+            aux_hash = calculate_aux_hash(aux_path)
+
+            stm_dll = stm_build_dir / "Stm.dll"
+            pcd = "PcdAuxBinHash"
+            pcd_addr = get_patch_pcd_address(stm_build_dir / "Stm.map", stm_dll, pcd)
+
+            if pcd_addr is None:
+                logging.error(f"PCD {pcd} not found in STM PCD table.")
+                return -1
+            
+            patch_pcd_value(stm_dll, pcd_addr, aux_hash)
+            generate_stm_binary(stm_dll, output_dir)
+
+            # Stm.bin and MmSuervisorCore.aux are already in the correct location
+            shutil.copy2(
+                mm_supervisor_build_dir / "MmSupervisorCore.efi",
+                output_dir / "MmSupervisorCore.efi"
+            )
+
+        except FileNotFoundError as e:
+            logging.error(f"File {e} not found.")
+            return 1
+        except RuntimeError as e:
+            logging.error(e)
+            return -1
+
+        return 0
+
+
+def generate_aux_file(aux_config_path: Path, mm_supervisor_build_dir: Path, output_dir: Path):
+    """Generates the auxiliary file for the MmsupervisorCore.
+   
+    Args:
+        aux_config_path: Path to the aux gen config file.
+        mm_supvervisor_build_dir: Path to the MM Supervisor build output.
+        output_dir: Path to place the artifacts.
+        
+    Raises:
+        RuntimeError: if gen_aux fails
+    """
+
+    output_path = output_dir / 'MmSupervisorCore.aux'
+
+    args = "run --"
+    args += f" --pdb {str(mm_supervisor_build_dir / 'MmSupervisorCore.pdb')}"
+    args += f" --efi {str(mm_supervisor_build_dir / 'MmSupervisorCore.efi')}"
+    args += f" --output {str(output_path)}"
+    args += f" --config {str(aux_config_path)}"
+
+    ret = RunCmd("cargo", args)
+    if ret != 0:
+        raise RuntimeError("gen_aux failed. Is your Cargo workspace setup correctly?")
+    
+    return output_path
+
+
+def calculate_aux_hash(file: Path):
+    """Calculates the hash of the auxiliary file.
+   
+    Args:
+        file: Path to the auxiliary file.
+        
+    Returns:
+        The hash of the auxiliary file.
+
+    Raises:    
+        FileNotFoundError: The file does not exist
+    """
+    if not file.exists():
+        raise FileNotFoundError(file)
+
+    hasher = hashlib.new("sha256")

```

Comment:
nit: Given this is hardcoded to sha256, we probably should note this more explicitly somewhere.

Reply:
Updated to mention the algorithm used is SHA256 in SpamPkg.dec, Stm.inf, and in GenSpamArtifacts.py